src/lib/ext-activitylog.js

```diff
@@ -4,25 +4,50 @@ class Model {
   constructor() {
     this.logs = [];
     this.filter = {
-      id: [],
-      viewType: [],
-      type: [],
+      id: new Set(),
+      viewType: new Set(),
+      type: new Set(),
+      keyword: '',
     };
   }
 
   addNewLogs(logs) {
     this.logs.push(...logs);
   }
 
-  addFilter({ logKey, valueEquals }) {
-    this.filter[logKey].push(valueEquals);
+  /**
+   * Set the value of any key in filter
+   * @param {object} updateFilterProps - It contains key and value to update
+   * the filter.
```

Comment:
@atiqueahmedziad I think that to make this jsdoc really useful you should actually spell what are the properties expected in that object, what is their type and what they represent, the current comment is quite generic.

As a "reader of the jsdoc" I would like to see explicitly that the expected properties are `id`, `viewType`, `type` and `keyword`, and then I would like to know what `id`, `viewType`, `type` and `keyword` represents and what is their expected types.